server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -757,6 +757,10 @@ private static Repository createRepository(
         }
     }
 
+    public Repository createRepository(RepositoryMetadata repositoryMetadata) {
```

Comment:
I think this deserves a bit of documentation: it relies on non internal factories, expects the repository to be closed after use, and warns about expert only usage in plugins/modules.

I also think we can replace some usages of the other `createRepository(RepositoryMetadata, factories, Function)` method with this one.